segmentstore/server/host/src/main/java/io/pravega/service/server/host/stat/AutoScaleProcessor.java

```diff
@@ -185,18 +186,29 @@ private void triggerScaleDown(String streamSegmentName, boolean silent) {
 
     private CompletableFuture<Void> writeRequest(ScaleEvent event) {
         CompletableFuture<Void> result = new CompletableFuture<>();
+        CompletableFuture<Void> writeComplete = new CompletableFuture<>();
         try {
-            CompletableFuture.runAsync(() -> {
+            AckFuture ackFuture = writer.get().writeEvent(event.getKey(), event);
+            ackFuture.addListener(() -> writeComplete.complete(null), executor);
+            writeComplete.thenAcceptAsync((Void v) -> {
```

Comment:
you could just pass this as the first argument in the line above. There is no need to double chain like this.